plugins/circleci/src/circleci-config.ts

```diff
@@ -293,33 +200,150 @@ const getInitialState = (): CircleCIState => {
   }
 }
 
+const rootOptionKeys = ['executors', 'jobs', 'workflows'] as const satisfies readonly (keyof Omit<
+  CircleCiOptions,
+  'custom'
+>)[]
+
 const isAutomatedConfig = (config: string): boolean => config.startsWith(automatedComment)
 
 const isNotToolKitConfig = (config: string): boolean => !config.includes('tool-kit')
 
-const getJobName = (job: Job): string => (typeof job === 'string' ? job : Object.keys(job)[0])
+const isObject = (val: unknown): val is Record<string, unknown> => isPlainObject(val)
```

Comment:
**thought**: we keep writing this predicate, presumably there's a type predicates library somewhere we can use?

Reply:
🦄 https://github.com/sindresorhus/is#objectvalue

Reply:
Herr Sorhus, you've done it again!